### ISSUE TITLE
Parse only throws first error for single field, if outer refine is present

### ISSUE THREAD

**Issue by aksdevac** [colinhacks/zod#480]
Consider Following Code:
```ts
const User = z
  .object({
    email: z
      .string()
      .min(1, { message: 'Email is required.' })
      .email({ message: 'Email is invalid.' }),
    password: z.string().min(8, { message: 'Password must have minimum 8 characters.' }),
    confirmPassword: z.string().min(1, { message: 'Confirm password required.' }),
  })
  .refine((data) => data.password === data.confirmPassword, {
    message: "Passwords don't match.",
    path: ['confirmPassword'], // path of error
  });

const parseResult = User2.parse({
  email: 'sds@',
  password: 'Ak111@202902',
  confirmPassword: 'Ak111@kj32l',
});

console.log({ parseResult });
```

Above code throws following error:
```js
[
  {
    "validation": "email",
    "code": "invalid_string",
    "message": "Email is invalid.",
    "path": [
      "email"
    ]
  }
]
```

**Expected Output:**
It should also return error for `Passwords don't match.`
Because of this issue all errors are not shown at once in `react-hook-form`.

**Versions**
Zod: 3.1.0

**Comment by Brodzko** [comment]
If this helps, I've added a `console.log` into my outer refinement and it isn't even registering so my guess would be the outer refinement isn't even running?

**Comment by aksdevac** [comment]
> If this helps, I've added a `console.log` into my outer refinement and it isn't even registering so my guess would be the outer refinement isn't even running?

Yes it is not running & not throwing error if any other field has error, so all errors are not shown at once Or onBlur, like in registration form user agrees to terms by clicking checkbox after entering confirm password, so until user checks the checkbox 'passwords don't match' error won't appear. So in my case user can see password don't match error only on Submit click, after agreeing to terms.

**Comment by tparvi** [comment]
This seems like exactly the same issue I am running into: https://github.com/colinhacks/zod/discussions/511 and there is codesandbox to reproduce this https://codesandbox.io/s/zod-refine-array-nh5b6

I'm trying to use zod with react-hook-form but there is simple validate button that calls just zod. I wanted to rule out possible react-hook-form/zod integration issue

**Comment by kutsan** [comment]
@colinhacks Thanks, that was fast. Upgrade to 3.10 and I'm not having the problem anymore.
